pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -339,6 +376,66 @@ public static int parseBlockComment(final char[] query, int offset) {
   }
 
   /**
+   * Parse string to check presence of INSERT keyword.
+   * @param query char[] of the query statement
+   * @param offset position of query to start checking
+   * @param isUpper is the text expected to be upper/lower case
+   * @return boolean indicates presence of word
+   */
+  public static boolean parseInsertKeyword(final char[] query, int offset, boolean isUpper) {
+    if (query.length < (offset + 7)) {
+      return false;
+    }
+
+    if (isUpper && query[offset] == 'I' && query[offset + 1] == 'N' && query[offset + 2] == 'S' && query[offset + 3] == 'E' && query[offset + 4] == 'R' && query[offset + 5] == 'T' ) {
```

Comment:
I'll re-jig the code to use equalsIgnoreCase and a single method for each keyword.